build.sh

```diff
@@ -340,127 +339,158 @@ if [[ "$DO_UPDATE" == true || ! -f "$HTML_CACHE/w3cbugs.csv" ]]; then
-}
+  }
 
-runWattsi "$HTML_TEMP/source-whatwg-complete" "$HTML_TEMP/wattsi-output"
-if [[ "$WATTSI_RESULT" == "0" ]]; then
-  if [[ "$LOCAL_WATTSI" != true ]]; then
-    "$QUIET" || grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
-  fi
-else
-  if [[ "$LOCAL_WATTSI" != true ]]; then
-    "$QUIET" || grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
-  fi
-  if [[ "$WATTSI_RESULT" == "65" ]]; then
-    echo
-    echo "There were errors. Running again to show the original line numbers."
-    echo
-    runWattsi "$HTML_SOURCE/source" "$HTML_TEMP/wattsi-raw-source-output"
+  runWattsi "$HTML_TEMP/source-whatwg-complete" "$HTML_TEMP/wattsi-output"
+  if [[ "$WATTSI_RESULT" == "0" ]]; then
+    if [[ "$LOCAL_WATTSI" != true ]]; then
+      "$QUIET" || grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
+    fi
+  else
     if [[ "$LOCAL_WATTSI" != true ]]; then
-      grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
+      "$QUIET" || grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
     fi
+    if [[ "$WATTSI_RESULT" == "65" ]]; then
+      echo
+      echo "There were errors. Running again to show the original line numbers."
+      echo
+      runWattsi "$HTML_SOURCE/SOURCE_LOCATION" "$HTML_TEMP/wattsi-raw-source-output"
+      if [[ "$LOCAL_WATTSI" != true ]]; then
+        grep -v '^$' "$HTML_TEMP/wattsi-output.txt" # trim blank lines
+      fi
+    fi
+    echo
+    echo "There were errors. Stopping."
+    exit "$WATTSI_RESULT"
   fi
-  echo
-  echo "There were errors. Stopping."
-  exit "$WATTSI_RESULT"
-fi
 
-perl .post-process-partial-backlink-generator.pl "$HTML_TEMP/wattsi-output/index-html" > "$HTML_OUTPUT/index.html";
-
-cp -p  entities/out/entities.json "$HTML_OUTPUT"
-cp -p "$HTML_TEMP/wattsi-output/xrefs.json" "$HTML_OUTPUT"
-
-# multipage setup
-rm -rf "$HTML_OUTPUT/multipage"
-mv "$HTML_TEMP/wattsi-output/multipage-html" "$HTML_OUTPUT/multipage"
-mv "$HTML_TEMP/wattsi-output/multipage-dev" "$HTML_OUTPUT/dev"
-rm -rf "$HTML_TEMP"
-
-echo "User-agent: *
-Disallow: /commit-snapshots/" > "$HTML_OUTPUT/robots.txt"
-cp -p  "$HTML_SOURCE/404.html" "$HTML_OUTPUT"
-cp -p "$HTML_SOURCE/link-fixup.js" "$HTML_OUTPUT"
-cp -p "$HTML_SOURCE/html-dfn.js" "$HTML_OUTPUT"
-cp -pR "$HTML_SOURCE/fonts" "$HTML_OUTPUT"
-cp -pR "$HTML_SOURCE/images" "$HTML_OUTPUT"
-cp -pR "$HTML_SOURCE/demos" "$HTML_OUTPUT"
-cp -pR "$HTML_SOURCE/dev" "$HTML_OUTPUT"
+  if [[ "$BUILD_TYPE" == "default" ]]; then
+    # Singlepage HTML
+    perl .post-process-partial-backlink-generator.pl "$HTML_TEMP/wattsi-output/index-html" > "$HTML_OUTPUT/index.html";
+
+    # Singlepage Commit Snapshot
+    COMMIT_DIR="$HTML_OUTPUT/commit-snapshots/$HTML_SHA"
+    mkdir -p "$COMMIT_DIR"
+    perl .post-process-partial-backlink-generator.pl "$HTML_TEMP/wattsi-output/index-snap" > "$COMMIT_DIR/index.html";
+
+    cp -p  entities/out/entities.json "$HTML_OUTPUT"
+    cp -p "$HTML_TEMP/wattsi-output/xrefs.json" "$HTML_OUTPUT"
```

Comment:
I suppose at some point it may be a good idea to give the snapshots their own copy of all the supporting resources, here and below. But that can be a followup.

Reply:
Yeah, I was wondering about that. We'll also need to change the source document for that though as currently it assumes everything is relative to the root. And not doing that will require changes to multipage deployments.